api/versionedsignedproposal.go

```diff
@@ -29,7 +30,43 @@ type VersionedSignedProposal struct {
 	Altair    *altair.SignedBeaconBlock
 	Bellatrix *bellatrix.SignedBeaconBlock
 	Capella   *capella.SignedBeaconBlock
-	Deneb     *apiv1deneb.SignedBlockContents
+	Deneb     *deneb.SignedBlockContents
+}
+
+// Slot returns the slot of the signed proposal.
+func Slot(p VersionedSignedProposal) (phase0.Slot, error) {
+	switch p.Version {
+	case spec.DataVersionPhase0:
+		if p.Phase0 == nil || p.Phase0.Message == nil {
+			return 0, errors.New("no phase0 block")
+		}
```

Comment:
If the code for the case is present then you may as well return the slot number, even though it's unlikely to ever be hit.  Or you can remove the case entirely, I don't mind which.

Reply:
yes i agree, phase0 code is very unlikely to get a hit. Removing the switch case for `DataVersionPhase0`